src/Catty/DataModel/Project/Project.m

```diff
@@ -45,104 +45,19 @@ - (instancetype)init
     return self;
 }
 
-- (NSInteger)numberOfTotalObjects
-{
-    return [self.objectList count];
-}
-
-- (NSInteger)numberOfBackgroundObjects
-{
-    NSInteger numberOfTotalObjects = [self numberOfTotalObjects];
-    if (numberOfTotalObjects < kBackgroundObjects) {
-        return numberOfTotalObjects;
-    }
-    return kBackgroundObjects;
-}
-
-- (NSInteger)numberOfNormalObjects
-{
-    NSInteger numberOfTotalObjects = [self numberOfTotalObjects];
-    if (numberOfTotalObjects > kBackgroundObjects) {
-        return (numberOfTotalObjects - kBackgroundObjects);
-    }
-    return 0;
-}
-
-- (SpriteObject*)addObjectWithName:(NSString*)objectName
-{
-    SpriteObject *object = [[SpriteObject alloc] init];
-    //object.originalSize;
-    object.spriteNode.currentLook = nil;
-
-    object.name = [Util uniqueName:objectName existingNames:[self allObjectNames]];
-    object.project = self;
-    [self.objectList addObject:object];
-    [self saveToDiskWithNotification:YES];
-    return object;
-}
-
-- (void)removeObjectFromList:(SpriteObject*)object
-{
-    // do not use NSArray's removeObject here
-    // => if isEqual is overriden this would lead to wrong results
-    NSUInteger index = 0;
-    for (SpriteObject *currentObject in self.objectList) {
-        if (currentObject == object) {
-            [currentObject removeSounds:currentObject.soundList AndSaveToDisk:NO];
-            [currentObject removeLooks:currentObject.lookList AndSaveToDisk:NO];
-            [currentObject.userData removeAllVariables];
-            [currentObject.userData removeAllLists];
-            currentObject.project = nil;
-            [self.objectList removeObjectAtIndex:index];
-            break;
-        }
-        ++index;
-    }
-}
-
-- (void)removeObject:(SpriteObject*)object
-{
-    [self removeObjectFromList:object];
-    [self saveToDiskWithNotification:YES];
-}
-
-- (void)removeObjects:(NSArray*)objects
-{
-    for (id object in objects) {
-        if ([object isKindOfClass:[SpriteObject class]]) {
-            [self removeObject:((SpriteObject*)object)];
-        }
-    }
-    [self saveToDiskWithNotification:YES];
-}
-
-- (BOOL)objectExistsWithName:(NSString*)objectName
-{
-    for (SpriteObject *object in self.objectList) {
-        if ([object.name isEqualToString:objectName]) {
-            return YES;
-        }
-    }
-    return NO;
-}
-
 #pragma mark - Custom getter and setter
-- (NSMutableArray<SpriteObject*>*)objectList
+- (Scene *)scene
```

Comment:
Please remove this getter as a preparation for further changes (replacement of one scene with several scenes).